docker_services_cli/services.py

```diff
@@ -8,30 +8,150 @@
 """Services module."""
 
 import subprocess
+import time
 
-from .config import DOCKER_SERVICES_FILEPATH
+import click
 
+from .config import DOCKER_SERVICES_FILEPATH, MYSQL, POSTGRESQL
 
-def services_down(filepath=DOCKER_SERVICES_FILEPATH):
-    """Stops the given services.
 
-    It does not requries the services. It stops containers and removes
-    containers, networks, volumes, and images created by ``up``.
+def _run_health_command(command):
+    """Runs a given command, returns True if it succeeds, False otherwise."""
+    try:
+        subprocess.check_call(
+            command,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL
+        )
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+def es_healthcheck(*args, **kwargs):
+    """Elasticsearch healthcheck."""
+    return _run_health_command([
+        "curl",
+        "-f",
+        "localhost:9200/_cluster/health?wait_for_status=green"
+    ])
+
+
+def postgresql_healthcheck(*args, **kwargs):
+    """Postgresql healthcheck."""
+    filepath = kwargs['filepath']
+
+    return _run_health_command([
+        "docker-compose",
+        "--file",
+        filepath,
+        "exec",
+        "postgresql",
+        "bash",
+        "-c",
+        "pg_isready",
+    ])
+
+
+def mysql_healthcheck(*args, **kwargs):
+    """Mysql healthcheck."""
+    filepath = kwargs['filepath']
+    password = MYSQL["MYSQL_ROOT_PASSWORD"]
+
+    return _run_health_command([
+        "docker-compose",
+        "--file",
+        filepath,
+        "exec",
+        "mysql",
+        "bash",
+        "-c",
+        f"mysql -p{password} -e \"select Version();\"",
+    ])
+
+
+def redis_healthcheck(*args, **kwargs):
+    """Redis healthcheck."""
+    filepath = kwargs['filepath']
+
+    return _run_health_command([
+        "docker-compose",
+        "--file",
+        filepath,
+        "exec",
+        "redis",
+        "bash",
+        "-c",
+        "redis-cli ping",
+        "|",
+        "grep 'PONG'",
+        "&>/dev/null;",
+    ])
+
+
+# FIXME: This should be moved to config.py
+# However, is not trivial due to possible ciruclar dependencies
+# Importing a string is not trivial without depending on invenio-base.utils
+# or duplicating the loading code.
+READYNESS_CHECKS = {
+    "es": es_healthcheck,
+    "postgresql": postgresql_healthcheck,
+    "mysql": mysql_healthcheck,
+    "redis": redis_healthcheck,
+}
+"""Readyness check functions module path, as string."""
```

Comment:
In my opinion, I think that this code belongs to the `wait_for_services` implementation, it is just what in other languages would be a switch/case etc.. so no need for the comment.

Reply:
Fine. I am still not convinced because by that line of thought we should have the services configuration (e.g. `MYSQL`) also in the `services.py` file.

In any case to be able to keep going I opened an question issue <https://github.com/inveniosoftware/docker-services-cli/issues/4> and removed the comment.